server/app.test.js

```diff
@@ -0,0 +1,68 @@
+const supertest = require('supertest');
+const app = require('./app.js');
+const request = supertest(app);
+
+jest.mock('../database/models/room.js');
+const {findByID} = require('../database/models/room.js');
+
+describe('server', () => {
+
+  describe('requests to /', () => {
+    test('it should respond to a GET request', (done) => {
+      request.get('/').then((response) => {
+        expect(response.statusCode).toBe(200);
+        done();
+      });
+    });
+
+    test('it should respond with the index.html', (done) => {
+      request.get('/').then((response) => {
+        expect(response.text).toContain('id="app"');
+        done();
+      });
+    });
+
+    test('it should respond to a POST request with 404', (done) => {
+      request.post('/').then((response) => {
+        expect(response.statusCode).toBe(404);
```

Comment:
If this is meant as a rejection, this should be status code 401 as convention for 'unauthorized' use